lib/active_snapshot/models/concerns/diffable.rb

```diff
@@ -0,0 +1,95 @@
+module ActiveSnapshot
+  module Diffable
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def diff(from:, to:)
+        if !from.is_a?(ActiveSnapshot::Snapshot) && !to.is_a?(ActiveSnapshot::Snapshot)
+          raise ArgumentError, "At least one of 'from' or 'to' must be an ActiveSnapshot::Snapshot"
+        end
+
+        from_item_id, from_item_type = from.is_a?(ActiveSnapshot::Snapshot) ? [from.item_id, from.item_type] : [from.id, from.class.name]
+        to_item_id, to_item_type = to.is_a?(ActiveSnapshot::Snapshot) ? [to.item_id, to.item_type] : [to.id, to.class.name]
+
+        if from_item_id != to_item_id || from_item_type != to_item_type
+          raise ArgumentError, "Both 'from' and 'to' must reference the same item (item_id and item_type must match)"
+        end
+
+        from_snapshot = from.is_a?(ActiveSnapshot::Snapshot) ? from : from.send(:build_snapshot)
+        to_snapshot = to.is_a?(ActiveSnapshot::Snapshot) ? to : to.send(:build_snapshot)
```

Comment:
@westonganger I'm not sure about this.
- I don't like calling a private method here
- It doesn't make sense to expose the `build_snapshot`
- I think it's very useful to be able to compare a snapshot to an instance 😂 

Let me know what you think

Reply:
We could probably make build_snapshot into a class method on the Snapshot class